WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddCustomPackageViewModel.swift

```diff
@@ -0,0 +1,110 @@
+import Foundation
+
+final class WooShippingAddCustomPackageViewModel: ObservableObject {
+    // Holds values for all dimension input fields.
+    // Using a dictionary so we can easily add/remove new types
+    // if needed just by adding new case in enum
+    @Published var fieldValues: [WooShippingPackageDimensionType: String] = [:]
+    // Holds selected package type when custom package is selected, it can be `box` or `envelope`
+    @Published var packageType: WooShippingPackageType = .box
+    // Holds value for toggle that determines if we are showing button for saving the template
+    @Published var showSaveTemplate: Bool = false
+    @Published var packageTemplateName: String = ""
+
+    // Field values are invalid if one of them is empty
+    var areFieldValuesInvalid: Bool {
+        for (_, value) in fieldValues {
+            if value.isEmpty {
+                return true
+            }
+        }
+        return fieldValues.count != WooShippingPackageDimensionType.allCases.count
+    }
+
+    func clearFieldValues() {
+        fieldValues.removeAll()
+    }
+
+    func resetValues() {
+        clearFieldValues()
+        showSaveTemplate = false
+        packageTemplateName = ""
+    }
```

Comment:
Should this also reset the `packageType` to `.box`? I could see it going either way (remembering the previously used package type or resetting it), but just asking because I noticed that if I select the envelope type that isn't reset.